packages/labs/virtualizer/src/test/scenarios/element-and-directive-parity.test.ts

```diff
@@ -148,6 +148,11 @@ describe('lit-virtualizer and virtualize directive', () => {
     expect(ulv.shadowRoot?.textContent).to.include('[5 selected]');
     expect(uvd.shadowRoot?.textContent).to.include('[5 selected]');
 
+    // There may occasionally be visibility change events from the initial render, so
+    // we clear those out before we start watching for new ones to avoid flaky tess.
+    ulv.visibilityChangedEvents.length = 0;
+    uvd.visibilityChangedEvents.length = 0;
```

Comment:
Maybe I'm not reading this correctly but I don't see any assertion against `ulv.visibilityChangedEvents.length` between this newly added emptying and the next emptying in line 182 below. So the lack of this couldn't have been the source of the failed assertion on line 206, could it?

Reply:
It's not an assertion exactly, it's this bit on line 178-179

```ts
    await until(() => ulv.visibilityChangedEvents.length > 0);
    await until(() => uvd.visibilityChangedEvents.length > 0);
```

Those lines wait for the events to come in.  By truncating the collectors prior to these lines we ensure we actually get the initia visibility changed events and THEN clear them out to ensure we don't get any others after that ~ line 206

Reply:
The weird thing about the await untils is they almost operate like an assertion in the sense that if they time out the test fails, so it's worth reading them almost like an assertion.

Reply:
Aha! So if there's already stuff in there we'd not actually wait for the thing to happen. Thanks for the clarification!